frontend/src/js/components/settings/addonselection.js

```diff
@@ -11,77 +11,160 @@
         </p>
-        {relevantAddons.map(addOn => {
-          const isEligible = addOn.eligible.indexOf(updatedPlan) > -1;
-
-          return (
-            <div
-              key={addOn.name}
-              className={`planPanel ${classes.planPanel} addon ${isUpgrade ? 'upgrade' : ''} ${addOn.isEnabled ? 'active' : ''} ${isEligible ? '' : 'muted'}`}
-              onClick={e => (isEligible ? onAddOnClick(e, addOn.name, !addOn.isEnabled) : () => false)}
-            >
-              <Checkbox disabled={!isEligible} checked={addOn.isEnabled} />
-              <div className="bold">{addOn.title}</div>
-              {isUpgrade && (
-                <div className="flexbox column">
-                  <div className={`link-color bold ${classes.price}`}>{addOn[updatedPlan].price}</div>
-                  <div>{addOn[updatedPlan].deviceCount}</div>
-                </div>
-              )}
-              <InfoText variant="dense">{addOn.description}</InfoText>
-              {/* eslint-disable-next-line react/jsx-no-target-blank */}
-              <a className="margin-left-small" href="https://mender.io/plans/features" target="_blank" rel="noopener">
-                Learn more
-              </a>
-            </div>
-          );
-        })}
+        <div className="flexbox column">
+          {relevantAddons.map(addOn => {
+            const isEligible = addOn.eligible.indexOf(currentPlan) > -1;
+            return (
+              <div key={addOn.name} className={`${classes.planPanel} addon ${addOn.isEnabled ? 'active' : ''} ${isEligible ? '' : 'muted'}`}>
+                <a
+                  className={`${classes.link} flexbox center-aligned bold`}
+                  href={`https://mender.io/pricing/add-ons/${addOn.name}`}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  Mender {addOn.title}
+                  <LaunchIcon className="link-color margin-left-x-small" fontSize="small" />
+                </a>
+                {isUpgrade && currentPlanName !== 'Enterprise' ? (
+                  <div className={classes.price}>
+                    {isEligible ? (
+                      <>
+                        starting <span className="bold">{addOn[currentPlan].price}</span>
+                      </>
+                    ) : (
+                      <>not available on {currentPlanName} plan</>
+                    )}
+                  </div>
+                ) : (
+                  currentPlanName === 'Enterprise' && <div />
+                )}
+                {addOn.isEnabled ? (
+                  <>
+                    <div className={classes.chip}>active</div>
+                    {addOn.pending && !addOn.pending.isAdd ? (
+                      <div className={`${classes.chip} margin-left-small`}>removal pending</div>
+                    ) : (
+                      <RemoveButton disabled={!isEligible} onClick={e => onAddOnClick(e, addOn.name, false)}>
```

Comment:
```suggestion
                      <RemoveButton disabled={!isEligible} onClick={e => onAddOnClick(addOn.name, false)}>
```
as you're no longer using the `e`